### PR TITLE
Add type definition for datadog-statsd-metrics-collector 

### PR DESCRIPTION
**Pull request by xzyfer**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.

### CODE REVIEW

File: types/datadog-statsd-metrics-collector/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// Type definitions for datadog-statsd-metrics-collector 1.1
+// Project: https://github.com/xzyfer/datadog-statsd-metrics-collector#readme
+// Definitions by: Michael Mifsud <https://github.com/xzyfer>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import dogstatsd = require('node-dogstatsd');
+
+export default class Collector implements dogstatsd.StatsDClient {
```
Comment (sandersn) [inline types/datadog-statsd-metrics-collector/index.d.ts]: datadog-statsd-metrics-collector uses `module.exports =` not `export default`, so the types should use the Typescript syntax for this: `export = Collector`. And imports should use `import Collector = require('datadog-statsd-metrics-collector');`

**Review by sandersn — CHANGES_REQUESTED** [review]

File: types/datadog-statsd-metrics-collector/index.d.ts (hunk 1 above)
Comment (xzyfer) [reply types/datadog-statsd-metrics-collector/index.d.ts]: Thanks. Feedback addressed.

Comment (xzyfer) [reply types/datadog-statsd-metrics-collector/index.d.ts]: Making this changes results in an error that I don't know how to resolve.

```
ERROR: 2:8  expect  TypeScript@next compile error: 
Module '"/home/travis/build/DefinitelyTyped/DefinitelyTyped/types/datadog-statsd-metrics-collector/index"' has no default export.
```

Comment (sandersn) [reply types/datadog-statsd-metrics-collector/index.d.ts]: You also have to change the import to the above syntax.

Comment (xzyfer) [reply types/datadog-statsd-metrics-collector/index.d.ts]: Thank you. This is now green.

**Review by sandersn — APPROVED** [review]

**Pull request merged**